mpas_analysis/shared/generalized_reader/generalized_reader.py

```diff
@@ -124,9 +126,31 @@ def open_multifile_dataset(fileNames, calendar, simulationStartTime=None,
                                  startDate=startDate,
                                  endDate=endDate)
 
-    ds = xarray.open_mfdataset(fileNames,
-                               preprocess=preprocess_partial,
-                               decode_times=False, concat_dim='Time', **kwargs)
+    kwargs = {'decode_times': False,
+              'concat_dim': 'Time'}
+
+    softLimit = resource.getrlimit(resource.RLIMIT_NOFILE)[0]
+
+    # use autoclose if we will use more than half the soft limit on open files
+    autoclose = len(fileNames) > softLimit/2
```

Comment:
The concept is good but I think the variable name isn't very intuitive.  A fraction would make more sense than a "scale factor" to me, so that's what I will probably add to `config.default`.  

We don't support multiple tasks yet, so as far as modifying the threshold for multiple tasks is concerned, it would be worth noting on #129 for addition at a later point.  I agree with the approach, though.